CHANGES.md

```diff
@@ -1,5 +1,13 @@
 # UNRELEASED
 
+## Language changes
+
+* `foreign` functions can now have an optional Cryptol implementation, which by
+  default is used when the foreign implementation cannot be found, or if the FFI
+  is unavailable. The `:set evalForeign` REPL option controls this behavior.
```

Comment:
I haven't looked much into the actual code, but what exactly does "when the foreign implementation cannot be found" mean? I ask since I remember getting a shared library for use with Cryptol's FFI to be a somewhat tricky process, and I worry that if a user screws this up, then Cryptol might silently switch to the non-FFI implementation of a function without the user realizing it. Is it obvious when a function's foreign implementation cannot be found?

Reply:
Yes, we emit a warning at load time. It looks something like
https://github.com/GaloisInc/cryptol/blob/5a7746799a9151cc4090e7ec17954c86d2afcc2b/tests/ffi/ffi-fallback.icry.stdout#L5-L8
or
https://github.com/GaloisInc/cryptol/blob/5a7746799a9151cc4090e7ec17954c86d2afcc2b/tests/ffi/ffi-fallback.icry.stdout#L100-L103

Reply:
Ah, I completely overlooked the `ffi-fallback.c` test case. That makes me feel much better, thanks.

Reply:
And to answer your first question, it is essentially when either `dlopen` fails (in which case we use the cryptol implementation for all foreign functions in the module) or one or more `dlsym` fails (in which case we use the cryptol implementation for the failed ones).